### ISSUE TITLE
-DOPTION_PRINT_SUPPORT=OFF causes build failure

### ISSUE THREAD

**Issue by OPNA2608** [fltk/fltk#98]
Noticed this while testing all the CMake options. Setting `OPTION_PRINT_SUPPORT` to `OFF` [as mentioned in the CMake README](https://github.com/fltk/fltk/blob/f09e17c3c564e8310125a10c03397cbf473ff643/README.CMake.txt#L166-L170) on Linux (NixOS 20.09.git) results in the following build error:

```
[ 16%] Building CXX object src/CMakeFiles/fltk.dir/Fl_Printer.cxx.o
/build/source/src/Fl_Printer.cxx:30:5: error: no declaration matches 'int Fl_Printer::begin_job(int, int*, int*)'
   30 | int Fl_Printer::begin_job(int pagecount, int *frompage, int *topage) {return 1;}
      |     ^~~~~~~~~~
In file included from /build/source/src/Fl_Printer.cxx:18:
/build/source/FL/Fl_Printer.H:94:7: note: candidate is: 'virtual int Fl_Printer::begin_job(int, int*, int*, char**)'
   94 |   int begin_job(int pagecount = 0, int *frompage = NULL, int *topage = NULL, char **perr_message = NULL);
      |       ^~~~~~~~~
/build/source/FL/Fl_Printer.H:86:17: note: 'class Fl_Printer' defined here
   86 | class FL_EXPORT Fl_Printer : public Fl_Paged_Device {
      |                 ^~~~~~~~~~
/build/source/src/Fl_Printer.cxx:67:5: error: no declaration matches 'int Fl_PostScript_File_Device::begin_job(int, int*, int*)'
   67 | int Fl_PostScript_File_Device::begin_job(int pagecount, int* from, int* to) {return 1;}
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /build/source/src/Fl_Printer.cxx:22:
/build/source/FL/Fl_PostScript.H:292:7: note: candidates are: 'int Fl_PostScript_File_Device::begin_job(FILE*, int, Fl_Paged_Device::Page_Format, Fl_Paged_Device::Page_Layout)'
  292 |   int begin_job(FILE *ps_output, int pagecount = 0, enum Fl_Paged_Device::Page_Format format = Fl_Paged_Device::A4,
      |       ^~~~~~~~~
/build/source/FL/Fl_PostScript.H:274:7: note:                 'int Fl_PostScript_File_Device::begin_job(int, Fl_Paged_Device::Page_Format, Fl_Paged_Device::Page_Layout)'
  274 |   int begin_job(int pagecount = 0, enum Fl_Paged_Device::Page_Format format = Fl_Paged_Device::A4,
      |       ^~~~~~~~~
/build/source/FL/Fl_PostScript.H:264:7: note:                 'virtual int Fl_PostScript_File_Device::begin_job(int, int*, int*, char**)'
  264 |   int begin_job(int pagecount, int* from, int* to, char **perr_message);
      |       ^~~~~~~~~
/build/source/FL/Fl_PostScript.H:248:17: note: 'class Fl_PostScript_File_Device' defined here
  248 | class FL_EXPORT Fl_PostScript_File_Device : public Fl_Paged_Device {
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [src/CMakeFiles/fltk.dir/build.make:720: src/CMakeFiles/fltk.dir/Fl_Printer.cxx.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:540: src/CMakeFiles/fltk.dir/all] Error 2
[…]
make: *** [Makefile:150: all] Error 2
```

[Full build log](https://github.com/fltk/fltk/files/4898047/fltk14.log)

**Comment by Albrecht-S** [comment]
Confirmed.

**Comment by Albrecht-S** [comment]
It looks like a new argument has not been added to the dummy methods used when print support is OFF. I suggest the patch [begin_job.diff.txt](https://github.com/fltk/fltk/files/4904080/begin_job.diff.txt) as a first step. but then fluid can't be linked with error message
```
undefined reference to `Fl_Printer::is_current()'
```
Giving up. Hope the patch helps...

**Comment by ManoloFLTK** [comment]
That should be now fixed in the git repository ( 2984fe6 ).
